app/models/image.rb

```diff
@@ -1,8 +1,7 @@
 class Image < ActiveRecord::Base
   has_many :attachments, as: :asset
   has_many :sources, through: :attachments
-  validates :file_base, presence: :true
-  validates :mime_type, presence: :true
+  validates :file_name, presence: :true
   validates :height, numericality: { only_integer: true }
   validates :width, numericality: { only_integer: true }
```

Comment:
The `:height` and `:width` validations here will cause the upload to fail if their value is `nil`.  We can solve this by adding `allow_blank: true`, so they would read:

`validates :height, numericality: { only_integer: true }, allow_blank: true`
`validates :width, numericality: { only_integer: true }, allow_blank: true`